posts/crud-pattern-in-react-implementation.md

```diff
@@ -1,217 +1,33 @@
 
+# Recap
 
-**Recap:** `Books` is a Base component that handles **CRUD** actions and also renders `BookTable` and `BookForm` components. It passes actions and data to the child components. We have introduced it in the previous post that can be found [here](/blog/crud-pattern-in-react-intro/).
+In the previous post <a href="/blog/crud-pattern-in-react-introduction">here</a>, we checked the structure of `Books` component that implements CRUD actions and passes these actions and data (retrieved from API) to its child components `BookTable` and `BookForm`. In this post, We will check implementation details of `BookTable` and `BookForm` components.
 
-# 1. `BookTable` Component
+# BookTable Component - Implementation
 
-In general, A Table view is used to provide a summary of the data. It can also have an interface (eg. button) to trigger *Update* or *Delete* action on the selected data record. The `BookTable` uses `Table` component to present book entities in a summary view. It triggers *Update* or *Delete* action using `showEditForm` and `performDelete` callback functions, passed as `props` by the `Books` component (parent). It also passes the data required for the actions (book object, book id) back to the parent component.
+A table view is generally used to provide summary of the data. It may also provide an interface (eg. button) to trigger _Update_ and _Delete_ actions on the selected data record. The `BookTable` component presents book entities in a summary view in table format. It triggers _Update_ and _Delete_ actions using `showEditForm` and `performDelete` callback functions, passed as `props` by the `Books` component (parent). It also passes the data required for the actions (book object, book id) back to the parent component.
 
-React supports **Unidirectional/One-Way Data Flow** only. It means, the data is passed from parent to child as `props` which are immutable arguments and not the other way around. Our actions (update, delete) are defined in the `Books` component (parent), but we need the data (selected book id or object) from `BookTable` component (child) to perform the actions. It is not possible to pass data back in *Unidirectional Data Flow*. So how do we *pass back* data from child to parent component? We are able to achieve it by having an **Inverse Data Flow**.
+React supports **Unidirectional/One-Way Data Flow** only. It means, the data is passed from parent to child as `props` which are immutable arguments and not the other way around. Our actions (update, delete) are defined in the `Books` component (parent), but we need the data (selected book id or object) from `BookTable` component (child) to perform the actions. It is not possible to pass data back in _Unidirectional Data Flow_. So how do we _pass back_ data from child to parent component? We are able to achieve it by having an **Inverse Data Flow**.
```

Comment:
No Bold / No Italic

Reply:
React supports Unidirectional/One-Way Data Flow only. It means, the data is passed from parent to child as `props` which are immutable arguments ~~and not the other way around~~. Our actions (`update`, `delete`) are defined in the `Books` component (parent), but we need the data (selected book `id` or object) from `BookTable` component (child) to perform these actions. We can achieve this by explicitly mapping  _Inverse Data Flow_ from `BookTable` component to `Books` component.